### PR TITLE
fix(firestore-send-email): added fallback for finding email templates

### PR DESCRIPTION
**Pull request by dackers86**
fixes: https://github.com/firebase/extensions/issues/209

### CODE REVIEW

File: firestore-send-email/functions/src/templates.ts [hunk 1]
```diff
@@ -110,6 +112,12 @@ export default class Templates {
     this.waits.forEach((wait) => wait());
   }
 
+  checkTemplateExists = async (name) => {
+    const query = this.collection.where("name", "==", name).get();
```
Comment (Ehesp) [inline firestore-send-email/functions/src/templates.ts:116]:
```js
return this.collection.where("name", "==", name).get().then(t => !t.empty);
```

File: firestore-send-email/functions/src/templates.ts [hunk 2]
```diff
@@ -110,6 +112,12 @@ export default class Templates {
     this.waits.forEach((wait) => wait());
   }
 
+  checkTemplateExists = async (name) => {
```
Comment (Ehesp) [inline firestore-send-email/functions/src/templates.ts:115]: Doesn't need to be `async`.

File: firestore-send-email/functions/src/templates.ts [hunk 3]
```diff
@@ -122,9 +130,16 @@ export default class Templates {
...
-      return Promise.reject(
-        new Error(`tried to render non-existent template '${name}'`)
-      );
+      //fallback, check if template does exist, results may be cached
+      checkingMissingTemplate(name);
+      const templateExists = await this.checkTemplateExists(name);
+
+      if (!templateExists)
+        return Promise.reject(
+          new Error(`tried to render non-existent template '${name}'`)
```
Comment (Ehesp) [inline firestore-send-email/functions/src/templates.ts:139]: `tried` -> `Tried`.

**Review by Ehesp — CHANGES_REQUESTED** [review]

**Review by Ehesp — APPROVED** [review]

**Comment by huangjeff5** [comment]
Can we remove the .js files for this extension?

**Pull request merged**